src/Magnum/Math/Geometry/Test/IntersectionBenchmark.cpp

```diff
@@ -0,0 +1,156 @@
+ * @param sphereCenter Sphere center
+ * @param radius Sphere radius
+ * @param origin Origin of the cone
+ * @param normal Cone normal
+ * @param angle Cone opening angle (0 < angle < pi).
+ *
+ * Returns `true` if the sphere intersects with the cone. */
+template<class T> bool sphereConeGT(
+        const Math::Vector3<T>& sphereCenter, const T radius,
+        const Math::Vector3<T>& origin, const Math::Vector3<T>& normal, const Math::Rad<T> angle) {
+    const Math::Vector3<T> diff = sphereCenter - origin;
+    const Math::Vector3<T> dir = diff.normalized();
+    const Math::Rad<T> halfAngle = angle/T(2);
+
+    /* Compute angle between normal and point */
+    const Math::Rad<T> actual = Math::acos(dot(normal, dir));
+
+    /* Distance from cone surface */
+    const T distanceFromCone = Math::sin(actual - halfAngle)*diff.length();
+
+    /* Either the sphere center lies in cone, or cone is max radius away from the cone */
+    return actual <= halfAngle || distanceFromCone <= radius;
+}
+
+template<class T>
+Math::Matrix4<T> coneViewFromCone(const Math::Vector3<T>& origin, const Math::Vector3<T>& normal) {
+    return Math::Matrix4<T>::lookAt(origin, origin + normal, Math::Vector3<T>::yAxis()).inverted();
+}
+
+typedef Math::Vector2<Float> Vector2;
+typedef Math::Vector3<Float> Vector3;
+typedef Math::Vector4<Float> Vector4;
+typedef Math::Matrix4<Float> Matrix4;
+typedef Math::Frustum<Float> Frustum;
+typedef Math::Deg<Float> Deg;
+typedef Math::Rad<Float> Rad;
+
+struct IntersectionBenchmark: Corrade::TestSuite::Tester {
+    explicit IntersectionBenchmark();
+
+    void sphereConeNaive();
+    void sphereCone();
+    void sphereConeView();
+
+    std::vector<std::pair<Vector3, Float>> _spheres;
+    std::vector<std::tuple<Vector3, Vector3, Rad>> _cones;
+};
+
+IntersectionBenchmark::IntersectionBenchmark() {
+    addBenchmarks({&IntersectionBenchmark::sphereConeNaive,
+                   &IntersectionBenchmark::sphereCone,
+                   &IntersectionBenchmark::sphereConeView}, 1000);
+
+    /* Generate random data for the benchmarks */
+    std::random_device rnd;
+    std::mt19937 g(rnd());
+    /* Position distribution */
+    std::uniform_real_distribution<float> pd(-10.0f, 10.0f);
+    /* Radius distribution */
+    std::uniform_real_distribution<float> rd(0.0f, 5.0f);
+    /* Cone angle distribution */
+    std::uniform_real_distribution<float> ad(1.0f, 179.0f);
+
+    _cones.reserve(4);
+    for(int i = 0; i < 4; ++i) {
+        _cones.emplace_back(Vector3{pd(g), pd(g), pd(g)},
+                Vector3{pd(g), pd(g), pd(g)}.normalized(),
+                Rad(Deg(ad(g))));
+    }
+
+    _spheres.reserve(100);
+    for(int i = 0; i < 100; ++i) {
+        _spheres.emplace_back(Vector3{pd(g), pd(g), pd(g)}, rd(g));
+    }
+}
+
+void IntersectionBenchmark::sphereConeNaive() {
+    volatile bool b = false;
+    CORRADE_BENCHMARK(10) {
+        for(auto& cone : _cones) {
+            for(auto& sphere : _spheres) {
```

Comment:
code golf (making lines shorter) -- maybe like this?

```cpp
CORRADE_BENCHMARK(10) for(auto& cone: _cones) for(auto& sphere: _spheres) {
    b = b ^ ...
}
```

Also, if you can, coding style -- `for(a: b)` instead of `for(a : b)`. There's never a space before `:`. Similarly in other places.